adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/AbstractVertxBasedMqttProtocolAdapter.java

```diff
@@ -453,6 +526,25 @@ private void onAuthenticationSuccess(final MqttEndpoint endpoint, final Device a
         metrics.incrementMqttConnections(authenticatedDevice.getTenantId());
     }
 
+    private Future<MessageConsumer> createCommandConsumer(final Device authenticatedDevice) {
+        return getCommandConnection().getOrCreateCommandConsumer(authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId(), (delivery, message) -> {
+            LOG.debug("command received for [{}] with delivery ", authenticatedDevice);
+            onCommandReceived(authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId(), delivery, message);
+        }, close -> {
+            LOG.debug("command receiver link closed remotely"); // TODO reopen?
+            getCommandConnection().closeCommandReceiver(authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId());
+            getCommandConnection().closeCommandResponseSenders(authenticatedDevice.getTenantId(), authenticatedDevice.getDeviceId());
```

Comment:
But why are all reponse senders implicitly closed when the command receiver link was closed (remotely)?
Is that resonable?